### DISCUSSION TITLE
Access data saved by react native in swift

### DISCUSSION THREAD

**Discussion by andredewaard** [react-native-async-storage/async-storage#538] [Q&A]
## You want to:  
Ask question

## Details:
I have an React native app which saves a key with Async storage. That key is used to send via the api to the backend to see if the user is logged in or not.
Now im creating a Share extension for the app and want to do that in native swift because the plugin for react native to add a share extension is outdated. 
I want to access the key saved in storage in swift so i can do a call to the backend with swift to see if a user is logged in and receive some data to populate the share extension. 
How can i do this?
Im new to swift / xcode development so i have not much experience with that.
I already checked this [doc](https://react-native-async-storage.github.io/async-storage/docs/advanced/brownfield) but dont know how to implement that.

Hope somebody can help me with this.

**Comment by sinclas** [comment]
Did you find a solution to this?   I have the same challenge.

**Comment by jamesdunay** [comment]
Yeah, bumping this. I'm not entirely sure that the Brownfield integration does what is needed. Seems like the wording of it means those delegate methods are called from the js layer, but I'm (and probably others) are trying to access the AsyncStorage from swift. 

Below is the best that I've got atm, but it doesnt' seem to know anything about the data stored on the JS side. I did read in one stack overflow that since extension is under a different bundle ID you'd need a way to point the AsyncStorage at your apps bundle Id instead. 

Please ping if you have a working example or anything to add. Thanks!

```  
let asyncStorage = RNCAsyncStorage()
asyncStorage.methodQueue.async {
      asyncStorage.multiGet(["groups"]) { response in
        print(response)
      }
    }
